src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -1746,6 +1746,19 @@ bool ApplicationManagerImpl::ManageMobileCommand(
   }
 
   if (message_type == mobile_apis::messageType::response) {
+    if (!message->isValid()) {
+      LOG4CXX_ERROR(logger_, "Mobile message response is invalid");
+      smart_objects::SmartObjectSPtr response =
+          MessageHelper::CreateNegativeResponse(
```

Comment:
@AKalinich-Luxoft 
I understand PR descriptions as following: in case there is result code out of request allowed code list - return GENERIC_ERROR instead of HMI result code, but keep all other data. However, CreateNegativeResponse doesn't forward any data back in response.
Is my understanding wrong?

Reply:
@dev-gh your understanding is correct, however I don't think we should keep all other data for response in this case

Reply:
@AKalinich-Luxoft 
"I don't think" doesn't sound like a requirement.

Reply:
@dev-gh requirement, which was linked for this defect APPLINK-16281:
In case an application sends a request and:
-> either unknown issue happened
-> either something went wrong,
SDL must:
respond with "GENERIC_ERROR, success:"false" to mobile application.